packages/neon-api/__integration__/transaction/validator.ts

```diff
@@ -0,0 +1,388 @@
+
+  test("autoFix", async () => {
+    const transaction = new TransactionBuilder({
+      validUntilBlock: 1
+    }).build();
+    const validator = new TransactionValidator(rpcClient, transaction);
+    const result = await validator.validateValidUntilBlock(true);
+    const {
+      valid,
+      suggestions: {
+        validUntilBlock: { fixed, prev, suggestion }
+      }
+    } = result;
+    expect(valid).toBeTruthy();
+    expect(fixed).toBeTruthy();
+    expect(prev).toBe(1);
+    expect(suggestion).toBeGreaterThan(1);
+    expect(transaction.validUntilBlock).toBe(suggestion);
+  });
+});
+
+describe("validateScript", () => {
+  test("valid", async () => {
+    const script = sc.createScript({
+      scriptHash: CONST.ASSET_ID.NEO,
+      operation: "name",
+      args: []
+    });
+    const transaction = new TransactionBuilder({
+      script
+    }).build();
+    const validator = new TransactionValidator(rpcClient, transaction);
+    const result = await validator.validateScript();
+    expect(result.valid).toBeTruthy();
+  });
+
+  test("invalid", async () => {
+    const script = sc.createScript({
+      scriptHash: CONST.ASSET_ID.NEO,
+      operation: "not_exist_method",
+      args: []
+    });
+    const transaction = new TransactionBuilder({
+      script
+    }).build();
+    const validator = new TransactionValidator(rpcClient, transaction);
+    const result = await validator.validateScript();
+    expect(result.valid).toBeFalsy();
+    expect(result.suggestions.script.message).toBeDefined();
+  });
+});
+
+describe("validateSystemFee", () => {
+  test("valid", async () => {
+    const script = sc.createScript({
+      scriptHash: CONST.ASSET_ID.NEO,
+      operation: "name",
+      args: []
+    });
+    const transaction = new TransactionBuilder({
+      script,
+      systemFee: 10
+    }).build();
+    const validator = new TransactionValidator(rpcClient, transaction);
+    const result = await validator.validateSystemFee();
+    expect(result.valid).toBeTruthy();
+  });
+
+  describe("invalid", () => {
+    test("not enough", async () => {
+      const script = sc.createScript({
+        scriptHash: CONST.ASSET_ID.NEO,
+        operation: "name",
+        args: []
+      });
+      const transaction = new TransactionBuilder({
+        script,
+        systemFee: 0
+      }).build();
+      const validator = new TransactionValidator(rpcClient, transaction);
+      const result = await validator.validateSystemFee();
+      expect(result.valid).toBeFalsy();
+    });
+
+    test("precision", async () => {
+      const script = sc.createScript({
+        scriptHash: CONST.ASSET_ID.NEO,
+        operation: "name",
+        args: []
+      });
+      const transaction = new TransactionBuilder({
+        script,
+        systemFee: 10.1
+      }).build();
+      const validator = new TransactionValidator(rpcClient, transaction);
+      const result = await validator.validateSystemFee();
+      expect(result.valid).toBeFalsy();
```

Comment:
Similarly, there should be a suggestion here?

Reply:
OK